src/common/common.c

```diff
@@ -289,7 +292,7 @@ void *OQS_MEM_checked_aligned_alloc(size_t alignment, size_t size) {
 	void *ptr = OQS_MEM_aligned_alloc(alignment, size);
 	if (ptr == NULL) {
 		fprintf(stderr, "Memory allocation failed\n");
-		abort();
+		return NULL; //abort();
```

Comment:
yes. Please let me know if you agree to remove the _checked_ functions. If so, I will go ahead and remove them.

Reply:
We could—but then we'll be left with a bunch of common-code functions that rely on the "checked" functions aborting if allocation fails. If this code is left unchanged, won't it lead to more crashes and/or memory corruption? If it needs to be changed to handle unchecked functions, is it not simpler to keep the "checked" functions rather than factor out the check everywhere?

Reply:
The caller must perform a NULL check before dereferencing the pointer. Additionally, in the case of a NULL check failure, the caller is responsible for releasing any allocated resources to prevent memory leaks.

Reply:
> The caller must perform a NULL check before dereferencing the pointer. Additionally, in the case of a NULL check failure, the caller is responsible for releasing any allocated resources to prevent memory leaks.

Sure—but those NULL checks are not currently present in the codebase, because the callers were relying on the `checked_malloc` functions. If we replace the `checked_malloc` functions with the unchecked versions in this PR, the NULL checks must also be added in this PR. For example [this code](https://github.com/open-quantum-safe/liboqs/pull/1926/files#diff-bb6ff52a8c0e7a4ee4104e6bd5bf779dad957ea1a751ff4b1d9bc5b0eb9f41f7R519-R522) will dereference NULL on malloc failure.

Reply:
so now done -- agree, @SWilson4 (and mark resolved)?